### ISSUE TITLE
Feature Request: Upload to AWS S3

### ISSUE THREAD

**Issue by enly1** [mkloubert/vs-deploy#14]
Hi - loving the extension - 

The sftp option is likely to provide me with the core of my needs, but another useful target to be supported natively might be aws s3 as an endpoint.

Expect very simple to add - would be great for tweaking cloudformation content then doing stack builds etc all from a friendly editor.

Would only need bucket and target and rely on the client aws configuration to operate, or could be more feature rich providing aws credential support and the like.

**Comment by mkloubert** [comment]
**Implemented since version 3.27.0!**

Example:

```json
{
    "deploy": {
        "targets": [
            {
                "type": "s3bucket",
                "name": "My S3 bucket",

                "bucket": "my-bucket"
            }
        ]
    }
}
```

You also have to setup an `credentials` file with your access keys. you should look at the [wiki](https://github.com/mkloubert/vs-deploy/wiki/target_s3bucket) to get to known how this works.

**Comment by enly1** [comment]
Really cool stuff and amazing turnaround ... one comment though might be a requirement to be able to specify the target folder ( prefix ) for the file to be sent to within the bucket.

i.e. my project folder/mystack.cfn.json -> s3://somebucket/cloudformation/mystack.cfn.json.

If this isn't available, is it safe to assume the prefix relating to the relative directory structure is used, or will it end up in the root of the bucket ?

**Comment by mkloubert** [comment]
All right! 👍 

One of the most important properties.

**Is implemented since version 3.28.0 and available now!**

```json
{
    "deploy": {
        "targets": [
            {
                "type": "s3bucket",
                "name": "My S3 bucket",

                "bucket": "my-bucket",
                "dir": "folder1/folder2"
            }
        ]
    }
}
```

**Comment by enly1** [comment]
Thanks - working nicely so far :-)
